File: 811.subdomain-visit-count.cpp
/*
 * @lc app=leetcode id=811 lang=cpp
 *
 * [811] Subdomain Visit Count
 */
#include <vector>
#include <string>
#include <unordered_map>
using namespace std;
// @lc code=start
class Solution
{
    unordered_map<string, int> record;

public:
    vector<string> subdomainVisits(vector<string> &cpdomains)
    {
        for (auto &&cpdomain : cpdomains)
        {
            int pos = cpdomain.find(' ');
            int num = stoi(cpdomain.substr(0, pos));
            record[cpdomain.substr(pos + 1)] += num;
            while ((pos = cpdomain.find('.', pos + 1)) != -1)
            {
                record[cpdomain.substr(pos + 1)] += num;
            }
        }
        vector<string> res;
        for (auto &&p : record)
        {
            res.push_back(to_string(p.second) + ' ' + p.first);
        }
        return res;
    }
};
// @lc code=end

File: 811.subdomain-visit-count.py
#
# @lc app=leetcode id=811 lang=python3
#
# [811] Subdomain Visit Count
#
from collections import Counter
from typing import *
# @lc code=start


class Solution:
    def subdomainVisits(self, cpdomains: List[str]) -> List[str]:
        record = Counter()
        for cpdomain in cpdomains:
            pos = cpdomain.find(' ')
            num = int(cpdomain[:pos])
            record[cpdomain[pos+1:]] += num
            while (pos := cpdomain.find('.', pos+1)) != -1:
                record[cpdomain[pos+1:]] += num
        res = []
        for cpdomain, num in record.items():
            res.append(str(num)+' '+cpdomain)
        return res

# @lc code=end
